2/contrib/s2i/run

```diff
@@ -235,8 +277,39 @@ if [ "$(ls /usr/lib/jenkins/*.hpi 2>/dev/null)" ]; then
           continue
         fi
 	echo "Linking ${FILENAME} RPM installed Jenkins plugins to ${JENKINS_HOME} ..."  
+        # note note a force link, to not override upgrades when not upgrading the image
         ln -s  $FILENAME ${JENKINS_HOME}/plugins/${basefilename}.jpi
    done
+   
+   # since this is rhel, if it is also the initial start of the image, we can remove the 
+   # the /usr/lib/jenkins/*.hpi link from /opt/openshift/plugins/*.jpi that 
+   # was done in the rhel7 dockefile; that link is done to assist our s2i
+   # extension scenario, so that the existing plugins are available during 
+   # the s2i assemble step (which precedes the s2i run step handled by this
+   # script); however, we must keep any plugins pulled in by s2i assemble, so
+   # we only remove files which are the same as /usr/lib/jenkins
+   #
+   # also, this will preserve the links if nothing has changed so that subsequent
+   # release to release migrations can work since the links will result in picking
+   # up the new content in /usr/lib/jenkins
+   if [ ! -e ${JENKINS_HOME}/configured ]; then
+      echo "Cleaning up links to base image plugins in /opt/openshift/plugins as part of this initial image start"
```

Comment:
this won't accomplish anything.  Every time the container starts from the image again, these files will be back.

Reply:
disagree ... as testing verified, if we don't do this here, then the install_plugin call below will remove the the linking between /var/lib/jenkins/plugins and /usr/lib/jenkins ... I am pretty sure removing this commit is not viable given the second commit (but I did not explicitly try that).

To reword your review summary comments, the complexity in managing the multitude of injection mechanisms for plugins makes it difficult to keep track.

Most likely there are short term (get the migration functional) vs. longer term (redo this whole mess) approaches we could employ here.

I at least do not think it is advisable to tackle the longer term approach.

But yeah, walking through some things face to face, and maybe watching the tests progress, might be necessary for this one given the undo complexity.

Reply:
OK confirmed, this block is needed for bringing up this image initially.  Otherwise, the plugins in /var/lib/jenkins/plugins are not linked to /usr/lib/jenkins.

The `install_plugins` call below does a force copy from /opt/openshift/plugins as part of possible additions.  However, in the case of a new install, there are links for all the pluings in /opt/openshift/plugins to assist the s2i extension scenario noted above.  Trying to confirm the s2i extension scenario did not add anything is essentially this same or similar logic in that path.  This way seemed more intuitive.

I can also confirm this block is *NOT* needed in the upgrade scenario (i.e. we brought up jenkins on a PV with the 3.6 image, then upgraded).